spring-integration-core/src/main/java/org/springframework/integration/aggregator/CorrelatingMessageBarrier.java

```diff
@@ -93,9 +93,7 @@ protected void handleMessageInternal(Message<?> message) {
 		synchronized (lock) {
 			this.store.addMessagesToGroup(correlationKey, message);
 		}
-		if (logger.isDebugEnabled()) {
-			logger.debug(String.format("Handled message for key [%s]: %s.", correlationKey, message));
-		}
+		logger.debug(() -> String.format("Handled message for key [%s]: %s.", correlationKey, message));
```

Comment:
There is one more in this class using `String.format()`.
I see your general comment about formats support for logging API.
I think we had discussion before that it is better to keep that `LogAccessor` contract as close to other logging frameworks as possible and don't overhead it with something what could cause confuse for target users. I mean that one `debugFormat()` and similar.
The point of this optimization on our side to avoid code flow extra complexity when we remove those `if (logger.isDebugEnabled()) {`.
On the other hand the point of `Supplier<? extends CharSequence>` contract to avoid an overhead for the final strings creation when we don't need to log that message.

See `LogMessage.format()` for that purpose otherwise: ` * Build a lazily formatted message from the given format string and argument.`

I think even this sentence could be reworked to that "lazy formatted" one instead of `Supplier` wrapper around standard `String.format()`...

Reply:
I get the idea but I'm not really sure I'm doing it right. Please take a look.
The `LogMessage` implements CharSequence so I'm passing `LogMessage`s to `LogAccessor.debug(CharSequence)` for lazily formatting instead of wrapping them in `Supplier`s.